salt/modules/rbenv.py

```diff
@@ -232,19 +234,21 @@ def install_ruby(ruby, runas=None):
     '''
     ruby = re.sub(r'^ruby-', '', ruby)
 
-    env = None
-    env_list = []
+    env = {}
+    build_env = None
 
     if __grains__['os'] in ('FreeBSD', 'NetBSD', 'OpenBSD'):
-        env_list.append('MAKE=gmake')
+        env['MAKE'] = 'gmake'
 
     if __salt__['config.get']('rbenv:build_env'):
-        env_list.append(__salt__['config.get']('rbenv:build_env'))
+        build_env = __salt__['config.get']('rbenv:build_env')
     elif __salt__['config.option']('rbenv.build_env'):
-        env_list.append(__salt__['config.option']('rbenv.build_env'))
+        build_env = __salt__['config.option']('rbenv.build_env')
 
-    if env_list:
-        env = ' '.join(env_list)
+    if isinstance(build_env, dict):
+        env.update(build_env)
+    elif isinstance(build_env, str):
+        env.update(dict(build_arg.split('=', 1) for build_arg in salt.utils.args.shlex_split(six.text_type(build_env))))
```

Comment:
Looks like the python 3 test/builds are failing on this line. Will have to try to find time to dig into why. I figured this line might be an issue with both versions and I was guessing on trying to use some of the builtins to help with that.

Reply:
okay, it errors when build_env is empty for some reason only in the python 3 builds but not python 2.
In the test suite we have
```
with patch.dict(rbenv.__salt__,
                            {'config.get': MagicMock(return_value='True')}):
```
Which if I understand correctly is just returning config.get with nothing.. just saying it worked. Either way, classic mistake of me testing with the condition I wanted but not with the condition that could exist of not wanting compile options. Will work on a fix. 
I suppose, another test should be added to test when config.get does return a build option, but I barely understand what I am writing here let alone test syntax.

Reply:
The issue was in the split when the test value for the config was just `True`.  There was no `=` to split on thus not creating the proper number of args for the dict build process.